server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -110,6 +112,27 @@
     public Mono<BlobId> save(BucketName bucketName, InputStream data) {
         Preconditions.checkNotNull(data);
 
+        return Mono.defer(() -> savingSrategySelection(bucketName, data));
+    }
+
+    private Mono<BlobId> savingSrategySelection(BucketName bucketName, InputStream data) {
```

Comment:
you can use FileBackedOutputStream instead

Reply:
That's an `InputStream`

Reply:
Yes, at this point, that's true. But the calling code created a file just before, right?
Instead of creating the file in a first place, you can create a FileBackedOutputStream and this optimization can go.

Reply:
I do not create file.

Reply:
s/savingSrategySelection/savingStrategySelection